packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+/**
+ * Internal dependencies
+ */
+import _CustomSelect from '../custom-select';
+import type { LegacyCustomSelectProps } from '../types';
+import { CustomSelectItem } from '..';
+
+function _LegacyCustomSelect( props: LegacyCustomSelectProps ) {
+	// Forward props + store from v2 implementation
+	const store = Ariakit.useSelectStore( {
+		async setValue( value ) {
+			if ( ! props.onChange ) return;
+
+			// Executes the logic in a microtask after the popup is closed.
+			// This is simply to ensure the isOpen state matches that in Downshift.
+			await Promise.resolve();
+			const state = store.getState();
+
+			const changeObject = {
+				selectedItem: {
+					// value will always be a string for the legacy component
+					name: value as string,
+					key: state.activeId as string,
```

Comment:
I believe that the `key` here is supposed to match the options's `key` (see this screenshot showing a log of the `onChange` function on the legacy component)

![Screenshot 2024-01-18 at 11 41 45](https://github.com/WordPress/gutenberg/assets/1083581/285fe8f7-96e4-4960-9277-94aeb4974d4e)


We should probably add unit tests to check the argument of the `onChange` function too.

Reply:
I added a couple tests for this: https://github.com/WordPress/gutenberg/pull/57902/commits/82a475e83f7fa5418a4dad44376ebae416f3fa6c

Hopefully, this was what you had in mind; otherwise, let me know how I can modify them. 

I could see how the second test could be considered redundant, but I added it as it's the most common way I've seen the `onChange` implemented for `CustomSelectControl` in the codebase.